src/Nox.Generator/Domain/Entity.template.cs

```diff
@@ -18,27 +19,27 @@ public partial class {{className}}:{{className}}Base
 /// <summary>
 /// Record for {{entity.Name}} created event.
 /// </summary>
-public record {{entity.Name}}Created({{entity.Name}} {{entity.Name}}) : IDomainEvent;
+public record {{entity.Name}}Created({{entity.Name}}Base {{entity.Name}}) : IDomainEvent;
```

Comment:
Is there a reason to add the Base class here?

Reply:
Yes, since we add raise domain event methods inside base class, "this" keyword means base class. 

_domainEvents.Add(new EntityCreated(this);

Reply:
The event needs to be raised with the actual entity that was created for example, so I would suggest something like this:

![image](https://github.com/NoxOrg/Nox.Generator/assets/8514193/95fe1e7d-4241-45ee-a837-6ce9cc146ad0)

this way a service implementation, like crypto cash,  can override the raiseinternal and extend the behavior, that is the reason why we have a base class with all the implementation

Reply:
I updated the implementation.